### PR TITLE
feature(main): fix registry find manifest logic

### PR DESCRIPTION
**Pull request by cuisongliu**
Signed-off-by: cuisongliu <cuisongliu@qq.com>

close https://github.com/labring/sealos/issues/1878

### CODE REVIEW

File: pkg/registry/cmd/flags.go [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+}
+
+func (opts *registrySaveResults) CheckAuth() map[string]types.AuthConfig {
+	if !file.IsExist(opts.registryPullRegistryDir) {
+		_ = os.MkdirAll(opts.registryPullRegistryDir, 0755)
+	}
+	cfg, err := registry.GetAuthInfo()
+	if err != nil {
+		logger.Error("auth info is error: %+v", err)
+		os.Exit(1)
```
Comment (fanux) [inline pkg/registry/cmd/flags.go:84]: return error instead exit

File: pkg/registry/cmd/save.go [hunk 2]
```diff
@@ -30,88 +30,91 @@ import (
...
+			var (
+				imageContext = "."
+			)
+			if len(args) != 0 {
+				imageContext = args[0]
+			}
+			images, err := buildimage.List(imageContext)
+			if err != nil {
+				logger.Error("get images list is error: %s", err.Error())
+				os.Exit(1)
```
Comment (fanux) [inline pkg/registry/cmd/save.go:97]: RunE and return error

File: pkg/registry/filter.go [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+
+type Filter struct {
+	Name         string
+	Tag          string
+	nameStrategy FilterStrategy
+	tagStrategy  FilterStrategy
+}
+
+func (f *Filter) Validate() error {
+	check := func(filter string, t FilterType) (FilterStrategy, error) {
```
Comment (fanux) [inline pkg/registry/filter.go:58]: Why not just use regular expressions

File: pkg/registry/list.go [hunk 4]
```diff
@@ -17,75 +17,91 @@ limitations under the License.
...
+	}
+
+	logger.Debug("filter name=%s,strategy=%s", filter.Name, filter.nameStrategy)
+	logger.Debug("filter tag=%s,strategy=%s", filter.Tag, filter.tagStrategy)
+
+	var repos []string
+	if filter.nameStrategy == FilterStrategyEquals {
+		repos = []string{filter.Name}
+	} else {
+		repos, _ = reg.Repositories(func(data []string) []string {
```
Comment (fanux) [inline pkg/registry/list.go:52]: ignore errors?

File: docs/4.0/docs/examples/sealctl-registry.md [hunk 5]
```diff
@@ -0,0 +1,226 @@
+# Using sealctl manager registry image
```
Comment (SignorMercurio) [inline docs/4.0/docs/examples/sealctl-registry.md:1]:
```suggestion
# Using sealctl to manage registry images
```

File: docs/4.0/docs/examples/sealctl-registry.md [hunk 6]
```diff
@@ -0,0 +1,226 @@
...
+
+## Image save to local registry dir
```
Comment (SignorMercurio) [inline docs/4.0/docs/examples/sealctl-registry.md:3]:
```suggestion
## Save images to local registry directory
```

File: docs/4.0/docs/examples/sealctl-registry.md [hunk 7]
```diff
@@ -0,0 +1,226 @@
...
+      --root string             storage root dir (default "/var/lib/containers/storage")
+      --runroot string          storage state dir (default "/run/containers/storage")
+      --show-path               enable show code path
+      --storage-driver string   storage-driver (default "overlay")
+      --storage-opt strings     storage driver option
+```
+
+
+
+#### How to using
```
Comment (SignorMercurio) [inline docs/4.0/docs/examples/sealctl-registry.md:32]: Please replace `How to using` with `Usage` globally.

```suggestion
#### Usage
```

File: docs/4.0/docs/examples/sealctl-registry.md [hunk 8]
```diff
@@ -0,0 +1,226 @@
+# Using sealctl manager registry image
+
+## Image save to local registry dir
+
+### default type
+
+default type can auto list images, using rootfs list images logic .ex charts,manifests,images dir.
```
Comment (SignorMercurio) [inline docs/4.0/docs/examples/sealctl-registry.md:7]: Not sure how to change this right now. Any clarification?

File: docs/4.0/docs/examples/sealctl-registry.md [hunk 9]
```diff
@@ -0,0 +1,226 @@
...
+      --storage-driver string   storage-driver (default "overlay")
+      --storage-opt strings     storage driver option
+```
+#### How to using
+
+```shell
+sealctl  registry  save raw --images nginx
+```
+
+## Status registry
```
Comment (SignorMercurio) [inline docs/4.0/docs/examples/sealctl-registry.md:69]:
```suggestion
## Check registry status
```

File: docs/4.0/docs/examples/sealctl-registry.md [hunk 10]
```diff
@@ -0,0 +1,226 @@
...
+
+need sealctl login or sealos login it.
```
Comment (SignorMercurio) [inline docs/4.0/docs/examples/sealctl-registry.md:71]:
```suggestion
To use this command, you need to `sealctl login` or `sealos login` first.
```

File: docs/4.0/docs/examples/sealctl-registry.md [hunk 11]
```diff
@@ -0,0 +1,226 @@
...
+      --root string             storage root dir (default "/var/lib/containers/storage")
+      --runroot string          storage state dir (default "/run/containers/storage")
+      --show-path               enable show code path
+      --storage-driver string   storage-driver (default "overlay")
+      --storage-opt strings     storage driver option
+```
+
+#### How to using
+```
+root@default-golang-0:~# sealctl  registry  status
```
Comment (SignorMercurio) [inline docs/4.0/docs/examples/sealctl-registry.md:97]: I think things like `root@default-golang-0:~#` does not provide useful information and should be replaced with a cleaner `$` just to show that we're in the command line. We should not simply copy from our terminal and paste it into the docs without any modification.

```suggestion
$ sealctl registry status
```

File: docs/4.0/docs/examples/sealctl-registry.md [hunk 12]
```diff
@@ -0,0 +1,226 @@
...
+      --debug                   enable debug logger
+      --root string             storage root dir (default "/var/lib/containers/storage")
+      --runroot string          storage state dir (default "/run/containers/storage")
+      --show-path               enable show code path
+      --storage-driver string   storage-driver (default "overlay")
+      --storage-opt strings     storage driver option
+```
+
+#### How to using
+```
```
Comment (SignorMercurio) [inline docs/4.0/docs/examples/sealctl-registry.md:96]:
```suggestion
```shell
```

File: docs/4.0/docs/examples/sealctl-registry.md [hunk 13]
```diff
@@ -0,0 +1,226 @@
...
+
+filter key support: name, tag 
+
+expressions:
+
+  `*val`: suffix 
+  `val*`: prefix
+  `value`: equals
+  `<none>`: empty
+  `*val*`: contains
```
Comment (SignorMercurio) [inline docs/4.0/docs/examples/sealctl-registry.md:163]:
```suggestion
### Filtering

Keys for `--filter`:
- `name`
- `tag`

Expressions:
- `*val`: suffix 
- `val*`: prefix
- `value`: equals
- `<none>`: empty
- `*val*`: contains
```

File: docs/4.0/docs/examples/sealctl-registry.md [hunk 14]
```diff
@@ -0,0 +1,226 @@
...
+      --runroot string          storage state dir (default "/run/containers/storage")
+      --show-path               enable show code path
+      --storage-driver string   storage-driver (default "overlay")
+      --storage-opt strings     storage driver option
+```
+
+### How to using
+
+```shell
+sealctl  registry  rmi 192.168.64.63:5000/kube-apiserver:v1.19.0 -n 192.168.64.63:5000
```
Comment (SignorMercurio) [inline docs/4.0/docs/examples/sealctl-registry.md:194]: We should either use:

<pre>
```shell
$ sealctl <commands> ...
```
</pre>

Or:

<pre>
```shell
sealctl <commands> ...
```
</pre>

And should not use them both.

File: docs/4.0/docs/examples/sealctl-registry.md [hunk 15]
```diff
@@ -0,0 +1,226 @@
...
+```
+
+## Prune registry
+
+Must be using in registry node, using binary deploy mode.Not support container deploy.
```
Comment (SignorMercurio) [inline docs/4.0/docs/examples/sealctl-registry.md:199]:
```suggestion
This command can only be run on a registry node, which used the binary deploy mode. Does not support nodes using container deploy mode.
```

**Review by SignorMercurio — CHANGES_REQUESTED** [review]

**Review by fanux — APPROVED** [review]

**Pull request merged**